### PR TITLE
Merge spike

### PR DESCRIPTION
**Pull request by oliverbarnes**
Rebasing our spike on [coopdevs/decidim-module-action_delegator](https://github.com/coopdevs/decidim-module-action_delegator), and adapting it for our needs.

I'm progressively cleaning up the files we don't need, while keeping some that illustrate how to override existing Decidim core functionality.

And I'm merging our own changes in piecemeal, integrating the way things like authentication are handled.

### CODE REVIEW

**Comment by oliverbarnes** [comment]
Merged in our spike results and did a rough cleanup. Nothing tested yet (neither manually nor programmatically) - just enough to get started making it actually work without needing to look back at the spike code.

**Comment by oliverbarnes** [comment]
Merged in our spike results and did a rough cleanup. Nothing tested yet (neither manually nor programmatically) - just enough to get started making it actually work without needing to look back at the spike code.

**Comment by oliverbarnes** [comment]
Screen Shot 2020-10-24 at 1 36 08 AM

delegation widget working for delegation creation and deletion 🎉 

ready for review.

File: Gemfile [hunk 1]
```diff
@@ -10,9 +10,9 @@ base_path = ""
 base_path = "../" if File.basename(__dir__) == "development_app"
 require_relative "#{base_path}lib/decidim/action_delegator/version"
```
Comment (jinjagit) [inline Gemfile:11]: Should the directories in this path be renamed to something like 'liquid_voting' & the path here edited accordingly?

**Review by jinjagit — APPROVED** [review]
This looks great, and like it was a lot of work.  :muscle: 
  
I added just one comment about a path in the Gemfile that still uses `action_delegator`.

File: Gemfile (hunk 1 above)
Comment (oliverbarnes) [reply Gemfile:11]: well spotted, need to change this and decidim-consultations -> decidim-proposals

Comment (oliverbarnes) [reply Gemfile]: done.

**Pull request merged**